### PR TITLE
Update YoutubeDl.php

### PR DESCRIPTION
**Pull request by hasnhasan**
Get download percentage

### CODE REVIEW

**Comment by norkunas** [comment]
Hey,

I don't see anything related with the percentage in this update.?

**Comment by hasnhasan** [comment]
:) 👍 

https://gist.github.com/hasnhasan/020dfea6683dfa0d4e5c42f26d2037e8#file-gistfile1-txt

**Comment by norkunas** [comment]
Does this really matter for you? Because this lib more likely used in browser than cli.

**Comment by hasnhasan** [comment]
Yes, it's important to me.
I show this information with socket.io with user friendly design.

**Comment by norkunas** [comment]
@hasnhasan this won't work if the output option is set as `'output' => '%(title)s.%(ext)s',`

File: src/YoutubeDl.php [hunk 1]
```diff
@@ -139,7 +139,10 @@ private function jsonDecode($data): array
 
     private function processDownload(Process $process): Video
     {
-        $videoData = $this->jsonDecode(trim($process->getOutput()));
+        $jsonOutput = str_replace(['"','%(ext)s'],['','info.json'],$this->options['output']);
+        $jsonOutput = file_get_contents($this->getDownloadPath().'/'.$jsonOutput);
```
Comment (norkunas) [inline src/YoutubeDl.php]: In master branch there is no `getDownloadPath` method, please use property directly

File: src/YoutubeDl.php [hunk 2]
```diff
@@ -139,7 +139,10 @@ private function jsonDecode($data): array
 
     private function processDownload(Process $process): Video
     {
-        $videoData = $this->jsonDecode(trim($process->getOutput()));
+        $jsonOutput = str_replace(['"','%(ext)s'],['','info.json'],$this->options['output']);
```
Comment (norkunas) [inline src/YoutubeDl.php]: This should handle all the cases explained in https://github.com/rg3/youtube-dl#output-template

**Comment by norkunas** [comment]
Handled in 30645d4

**Pull request closed without merging**